t/Net/DRI/Protocol/RRI.t

```diff
@@ -48,65 +48,94 @@ $R2 = $E1 . '<tr:transaction><tr:stid>' . $TRID . '</tr:stid><tr:result>success<
 
-# Contact check
-$R2 = $E1 . '<tr:transaction><tr:stid>' . $TRID . '</tr:stid><tr:result>success</tr:result><tr:data><contact:checkData><contact:handle>DENIC-99990-10240-BSP</contact:handle><contact:status>failed</contact:status></contact:checkData></tr:data></tr:transaction>' . $E2;
-$rc = $dri->contact_check($c);
+# Contact info (based on: https://member.secure.denic.de/index.php?eID=dumpFile&t=f&f=7282&token=9f9392f2d7dc61d8286afb772603e6c2f53adde2)
+$R2 = $E1 . '<tr:transaction><tr:stid>' . $TRID . '</tr:stid><tr:result>success</tr:result><tr:data><contact:infoData><contact:handle>DENIC-99995-GENREQ</contact:handle><contact:type>REQUEST</contact:type><contact:uri-template>http://denic.de/contact/{Ulabel}</contact:uri-template><contact:changed>2010-03-31T15:51:08+02:00</contact:changed></contact:infoData></tr:data></tr:transaction>' . $E2;
+$rc = $dri->contact_info($dri->local_object('contact')->srid('DENIC-99995-GENREQ'));
 isa_ok($rc, 'Net::DRI::Protocol::ResultStatus');
-is(defined($rc) && $rc->is_success(), 1, 'Contact successfully checked');
-is_string($R1, '<?xml version="1.0" encoding="UTF-8" standalone="yes"?><registry-request xmlns="http://registry.denic.de/global/1.0" xmlns:contact="http://registry.denic.de/contact/1.0"><contact:check><contact:handle>DENIC-99990-10240-BSP</contact:handle></contact:check></registry-request>', 'Check Contact XML correct');
-is($dri->get_info('exist', 'contact', 'DENIC-99990-10240-BSP'), 1, 'Contact exists');
+is($rc->is_success(), 1, 'Contact successfully queried (version 3.0)');
+is_string($R1, '<?xml version="1.0" encoding="UTF-8" standalone="yes"?><registry-request xmlns="http://registry.denic.de/global/3.0" xmlns:contact="http://registry.denic.de/contact/3.0"><contact:info><contact:handle>DENIC-99995-GENREQ</contact:handle></contact:info></registry-request>', 'Query Contact XML correct (version 3.0)');
+$c = $dri->get_info('self', 'contact', 'DENIC-99995-GENREQ');
+isa_ok($c, 'Net::DRI::Data::Contact::DENIC');
+is($c->type() . '|' . $c->uri_template(),
```

Comment:
This is an odd test? Why are you testing these two variables concatenated with | instead of testing separately?

Reply:
Oh, isee some older tests were done like that...

Reply:
Yep. Kept to how it was before :)